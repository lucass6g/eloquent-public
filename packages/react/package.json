{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
