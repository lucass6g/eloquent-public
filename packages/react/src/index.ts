export {Switch, Case, Default} from './components/conditional-switch'