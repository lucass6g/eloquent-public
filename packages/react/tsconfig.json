{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
