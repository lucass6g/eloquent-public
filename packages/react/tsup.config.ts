import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})