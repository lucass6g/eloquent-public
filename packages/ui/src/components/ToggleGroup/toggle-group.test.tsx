import { describe, it, expect } from "vitest";
import { ToggleGroup } from ".";
import { Bold, Italic, Underline } from "lucide-react";
import { render, screen } from "@testing-library/react";

const mockToggleGroupComponent = (
  <ToggleGroup.Root
    variant="default"
    size="default"
    disabled={false}
    type="single"
  >
    <ToggleGroup.Item value="bold" aria-label="Toggle bold">
      <Bold className="h-4 w-4" />
    </ToggleGroup.Item>
    <ToggleGroup.Item value="italic" aria-label="Toggle italic">
      <Italic className="h-4 w-4" />
    </ToggleGroup.Item>
    <ToggleGroup.Item value="strikethrough" aria-label="Toggle strikethrough">
      <Underline className="h-4 w-4" />
    </ToggleGroup.Item>
  </ToggleGroup.Root>
);

describe("ToggleGroup Component", () => {
  it("should render ToggleGroup component with default values", async () => {
    const component = render(mockToggleGroupComponent);
    const toggleGroup = screen.getByRole("group");
    const toggleGroupItem = screen.getByLabelText("Toggle bold");

    expect(toggleGroup.children.length).toBe(3);
    expect(toggleGroupItem?.getAttribute("aria-label")).toBe("Toggle bold");
    expect(component.baseElement).toMatchSnapshot();
  });
});
