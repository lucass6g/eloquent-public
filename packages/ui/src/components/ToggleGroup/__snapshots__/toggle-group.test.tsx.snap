// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToggleGroup Component > should render ToggleGroup component with default values 1`] = `
<body>
  <div>
    <div
      class="flex items-center justify-center gap-1"
      dir="ltr"
      role="group"
      style="outline: none;"
      tabindex="0"
    >
      <button
        aria-checked="false"
        aria-label="Toggle bold"
        class="inline-flex items-center justify-between rounded-[6px] text-sm font-medium transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground [&>svg]:mr-2 bg-transparent h-10 px-3 w-[90px]"
        data-radix-collection-item=""
        data-state="off"
        role="radio"
        tabindex="-1"
        type="button"
      >
        <svg
          class="lucide lucide-bold h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 12h9a4 4 0 0 1 0 8H7a1 1 0 0 1-1-1V5a1 1 0 0 1 1-1h7a4 4 0 0 1 0 8"
          />
        </svg>
      </button>
      <button
        aria-checked="false"
        aria-label="Toggle italic"
        class="inline-flex items-center justify-between rounded-[6px] text-sm font-medium transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground [&>svg]:mr-2 bg-transparent h-10 px-3 w-[90px]"
        data-radix-collection-item=""
        data-state="off"
        role="radio"
        tabindex="-1"
        type="button"
      >
        <svg
          class="lucide lucide-italic h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="19"
            x2="10"
            y1="4"
            y2="4"
          />
          <line
            x1="14"
            x2="5"
            y1="20"
            y2="20"
          />
          <line
            x1="15"
            x2="9"
            y1="4"
            y2="20"
          />
        </svg>
      </button>
      <button
        aria-checked="false"
        aria-label="Toggle strikethrough"
        class="inline-flex items-center justify-between rounded-[6px] text-sm font-medium transition-colors hover:bg-muted hover:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 data-[state=on]:bg-accent data-[state=on]:text-accent-foreground [&>svg]:mr-2 bg-transparent h-10 px-3 w-[90px]"
        data-radix-collection-item=""
        data-state="off"
        role="radio"
        tabindex="-1"
        type="button"
      >
        <svg
          class="lucide lucide-underline h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6 4v6a6 6 0 0 0 12 0V4"
          />
          <line
            x1="4"
            x2="20"
            y1="20"
            y2="20"
          />
        </svg>
      </button>
    </div>
  </div>
</body>
`;
